install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -494,20 +494,17 @@ data:
   {{- if ne (.Values.routingMode | default "tunnel") "tunnel" }}
     {{- fail (printf "RoutingMode must be set to tunnel when aksbyocni.enabled=true" )}}
   {{- end }}
-  routing-mode: "tunnel"
-  tunnel-protocol: "vxlan"
-{{- else if .Values.routingMode }}
-  routing-mode: {{ .Values.routingMode | quote }}
-{{- else }}
-  # Default case
-  routing-mode: "tunnel"
-  tunnel-protocol: "vxlan"
+  {{- if ne (.Values.tunnelProtocol | default "vxlan" ) "vxlan" }}
+    {{- fail (printf "TunnelProtocol must be set to vxlan when aksbyocni.enabled=true" )}}
+  {{- end }}
 {{- end }}
 
-{{- if .Values.tunnelProtocol }}
-  tunnel-protocol: {{ .Values.tunnelProtocol | quote }}
+{{- if not .Values.gke.enabled }}
+  routing-mode: {{ .Values.routingMode | default "tunnel" | quote }}
```

Comment:
nit: Would it make sense to define a `$fallbackRoutingMode` variable on the top of the file that defaults to "tunnel" but is set to "native" in the GKE case? I think that would be a bit easier to read

Reply:
In the configuration I proposed, there are only 2 things to check:
(1) Do I have `gke.enabled: true`? If so, the routing-mode is hardcoded to "native".
If not: (2) Do I have `routingMode` set explicitly? If so, routing-mode is that value, otherwise it is "tunnel".

I think introducing a variable would increase the mental overhead required to understand where the value actually comes from and does not provide much benefit when we only have that one "edge case"-y if-statement for GKE in which the `routing-mode` is specified directly.

Reply:
If you disagree, I can change it to the variable, you are the boss.

Reply:
Yet another possibility could be:

```
routing-mode: {{ .Values.routingMode | default (ternary "native" "tunnel" .Values.gke.enabled) | quote }}
```

Reply:
That would not use a variable and only have one place where the routing-mode is defined. I like it.

Reply:
That also looks good to me. Let's do that.

Reply:
Done, please resolve the conversation if it fits